### PR TITLE
[2.0] Collection of backports

### PR DESCRIPTION
**Pull request by nabobalis**
Backport of #5038 among others.

### CODE REVIEW

**Comment by nabobalis** [comment]
I need to fix some tests

**Review by dstansby — CHANGES_REQUESTED** [review]
I don't think we should be merging or releasing workflows (readthedocs) in this case that rely on git branches instead of released packages.

File: tox.ini [hunk 1]
```diff
@@ -77,6 +78,8 @@ deps =
     figure-!devdeps: astropy==4.0.1.post1
     figure-!devdeps: matplotlib==3.3.2
 
+    build_docs: git+https://github.com/Cadair/sphinx-gallery@skip_on_fail#egg=sphinx-gallery
+
```
Comment (Cadair) [inline tox.ini:82]: We don't technically need this here as the tox build runs with `-W` so the behaviour is unchanged with or without the branch.

**Comment by Cadair** [comment]
> I don't think we should be merging or releasing workflows (readthedocs in this case) that rely on git branches instead of released packages.

We are already using this on the 2.1 and master branches to prevent issues on RTD for PR previews and the actual release build following [a conversation in #sunpy-development:openastronomy.org](https://matrix.to/#/!JYqfIVJjWANcHnfktY:cadair.com/$16133960461201mMqai:cadair.com?via=cadair.com&via=matrix.org). 

I am mostly indifferent to if we backport it or not, but I just thought I would explain the backstory.

**Comment by nabobalis** [comment]
Since the PR has been merged, I have updated it to point to upstream sphinx gallery.

**Review by Cadair — APPROVED** [review]

**Pull request merged**